nni/tools/nnictl/common_utils.py

```diff
@@ -95,3 +97,23 @@ def generate_folder_name():
         temp_dir = generate_folder_name()
     os.makedirs(temp_dir)
     return temp_dir
+
+class SimpleFileLock(filelock.SoftFileLock):
+    def __init__(self, lock_file, timeout=-1, stale=10):
+        super(__class__, self).__init__(lock_file, timeout)
+        self._stale = stale
+
+    def __enter__(self):
+        count = 0
+        while True:
+            try:
+                if os.path.isfile(self._lock_file) and time.time() - os.stat(self._lock_file).st_mtime > self._stale:
+                    os.remove(self._lock_file)
+                self.acquire()
+                return self
+            except Exception:
+                print_warning('[{}] fail lock file, auto try again!'.format(count))
+                count += 1
+
+def get_file_lock(path: string, timeout=-1, stale=10):
+    return SimpleFileLock(path + '.lock', timeout=timeout, stale=stale)
```

Comment:
what is the purpose to set stale = 10 seconds?

Reply:
Setting `stale=10` is unreasonable. Maybe modify it to `stale=-1` is better? means it will seize the lock immediately.